sys/net/gnrc/netif/gnrc_netif.c

```diff
@@ -83,6 +87,10 @@ int gnrc_netif_create(gnrc_netif_t *netif, char *stack, int stacksize,
 
     res = thread_create(stack, stacksize, priority, THREAD_CREATE_STACKTEST,
                         _gnrc_netif_thread, (void *)netif, name);
+
+    /* wait for result of driver init */
+    mutex_lock(&_netif_init_done);
```

Comment:
I am wondering if we can re-use the `netif`'s own mutex (locked with `gnrc_netif_acquire()` for that?